tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/ProcessHelper.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Azure.Sdk.Tools.Cli.Helpers
+{
+    public class ProcessHelper
+    {
+        public static ProcessResult RunProcess(string command, string[] args, string workingDirectory)
+        {
+            var processStartInfo = new ProcessStartInfo
+            {
+                FileName = command,
+                WorkingDirectory = workingDirectory,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                RedirectStandardInput = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            foreach (var arg in args)
+            {
+                processStartInfo.ArgumentList.Add(arg);
+            }
+
+            var output = new StringBuilder();
+            int exitCode = -1;
+
+            using (var process = new Process())
+            {
+                process.StartInfo = processStartInfo;
+                process.OutputDataReceived += (sender, e) =>
+                {
+                    if (e.Data != null)
+                    {
+                        output.AppendLine(e.Data);
+                    }
+                };
+                process.ErrorDataReceived += (sender, e) =>
+                {
+                    if (e.Data != null)
+                    {
+                        output.AppendLine(e.Data);
+                    }
+                };
+
+                Console.Error.WriteLine($"Running command: {command} {string.Join(" ", args)} in {workingDirectory}");
```

Comment:
We should dependency inject `ILogger` instead of using console. Lmk if you need an example.

Reply:
Logger injected! The process helper is also registered for dependency injection now too.